### PR TITLE
Readme: note that config changes don't apply to already compiled code

### PR DESCRIPTION
**Pull request by Kage-Yami**
*Please write a short comment explaining your change (or "none" for internal only changes)*

changelog: added a note to the Readme that config changes don't apply to already compiled code

### CODE REVIEW

**Comment by Kage-Yami** [comment]
This is documenting what was (re-)discovered as part of #8145.

File: README.md [hunk 1]
```diff
@@ -155,6 +155,11 @@ cognitive-complexity-threshold = 30
 See the [list of lints](https://rust-lang.github.io/rust-clippy/master/index.html) for more information about which
 lints can be configured and the meaning of the variables.
 
+Note that configuration changes will not apply for code that has already been compiled and cached under `./target/`;
+for example, adding a new string to `doc-valid-idents` may still result in Clippy flagging that string. To be sure that
+any configuration changes are applied, you may want to delete your `./target/` directory and re-compile your crate from
```
Comment (llogiq) [inline README.md:160]: That should probably reference `cargo clean`.

**Review by llogiq — COMMENTED** [review]
Just a small suggestion for improvement, otherwise this looks good to merge.

File: README.md (hunk 1 above)
Comment (Kage-Yami) [reply README.md:160]: I had forgotten that `cargo clean` was a thing; I'll amend the suggestion.

Comment (Kage-Yami) [reply README.md]: I've amended this to say:

> ... you may want to run `cargo clean` and re-compile...

**Comment by llogiq** [comment]
Thank you! @bors r+ squash

**Comment by llogiq** [comment]
It appears that the test runners have some intermittent problems. I'll ask infra.

**Comment by llogiq** [comment]
The remark step itself was successful. It seems something's wrong with github's test infra.

**Comment by llogiq** [comment]
It seems that bors hasn't successfully pushed to master despite the tests being OK.

@bors retry

**Comment by llogiq** [comment]
Ok, this seems to be a github UI problem where the PR is not marked as merge despite being the commit being merged to master (I just pulled and checked).

**Pull request closed without merging**